core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1582,6 +1582,42 @@ class WorkspaceService(protected val userInfo: UserInfo,
       }
     }
 
+  def retrySubmission(workspaceName: WorkspaceName, submissionRetry: SubmissionRetry, submissionId: String): Future[RetriedSubmissionReport] = {
+    getWorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.write) flatMap { workspaceContext =>
+      dataSource.inTransaction { dataAccess =>
+        withSubmission(workspaceContext, submissionId, dataAccess) { submission =>
+          val newSubmission = Submission(submissionId = submissionId,
+            submissionDate = DateTime.now(),
+            submitter = submission.submitter,
+            submissionRoot = submission.submissionRoot,
+            methodConfigurationNamespace = submission.methodConfigurationNamespace,
+            methodConfigurationName = submission.methodConfigurationName,
+            submissionEntity = submission.submissionEntity,
+            workflows = submissionRetry.retryType.filterWorkflows(submission.workflows),
+            status = submission.status,
+            useCallCache = submission.useCallCache,
+            deleteIntermediateOutputFiles = submission.deleteIntermediateOutputFiles,
+            useReferenceDisks = submission.useReferenceDisks,
+            memoryRetryMultiplier = submission.memoryRetryMultiplier,
+            workflowFailureMode = submission.workflowFailureMode,
+            externalEntityInfo = submission.externalEntityInfo,
+            userComment = submission.userComment)
+
+          implicit val subStatusCounter = submissionStatusCounter(workspaceMetricBuilder(workspaceContext.toWorkspaceName))
+          implicit val wfStatusCounter = (status: WorkflowStatus) =>
+            if (config.trackDetailedSubmissionMetrics) Option(workflowStatusCounter(workspaceSubmissionMetricBuilder(workspaceContext.toWorkspaceName, UUID.fromString(submissionId)))(status))
+            else None
+
+          for {
+            retriedSub <- dataAccess.submissionQuery.create(workspaceContext, submission)
```

Comment:
could you break out the `subStatusCounter`, `wfStatusCounter`, and `dataAccess.submissionQuery.create` into a shared method used by both `retrySubmission` and `saveSubmission`? At minimum, add a code comment to both places that if we change how we capture metrics, we need to change it in both places